app/controllers/application_controller.rb

```diff
@@ -13,7 +13,17 @@ def sort_methods
     ]
   end
 
-  def set_cart
-    @cart = Cart.new(session[:cart])
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id] && User.exists?(id: session[:user_id])
+  end
+
+  def current_admin?
+    current_user && current_user.admin?
   end
+
+    private
+
+    def set_cart
+      @cart = Cart.new(session[:cart])
+    end
```

Comment:
Why is set_cart the only one in the private area? I'm not too clear on how to use the private area for the application controller, but I feel like we should have all the helpers in the same area.